diff --git a/modules/custom/beignet/src/Controller/BeignetController.php b/modules/custom/beignet/src/Controller/BeignetController.php
index 4d51992..0c52765 100644
--- a/modules/custom/beignet/src/Controller/BeignetController.php
+++ b/modules/custom/beignet/src/Controller/BeignetController.php
@@ -2,12 +2,12 @@
 
 namespace Drupal\beignet\Controller;
 
-use Symfony\Component\HttpFoundation\Response;
-
 class BeignetController
 {
     public function eatBeignets($number)
     {
-        return new Response(sprintf('NOM %s beignets', $number));
+        return [
+            '#title' => sprintf('NOM %s beignets', $number),
+        ];
     }
 }
