diff --git a/modules/custom/beignet/src/Controller/BeignetController.php b/modules/custom/beignet/src/Controller/BeignetController.php
index c042056..84b5642 100644
--- a/modules/custom/beignet/src/Controller/BeignetController.php
+++ b/modules/custom/beignet/src/Controller/BeignetController.php
@@ -4,10 +4,10 @@
 
 use Drupal\beignet\Service\BeignetBakery;
 use Drupal\Core\Config\ConfigFactoryInterface;
-use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Controller\ControllerBase;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
-class BeignetController implements ContainerInjectionInterface
+class BeignetController extends ControllerBase
 {
     private $beignetBakery;
 
@@ -25,6 +25,9 @@ public function eatBeignets($number)
     {
         $beignets = $this->beignetBakery->bakeBeignets($number);
 
+        $this->getLogger('database')
+            ->info(sprintf('Eating %s beignets', count($beignets)));
+
         return [
             '#title' => 'NOM the following beignets: '.implode(', ', $beignets),
         ];
