diff --git a/modules/custom/beignet/src/Controller/BeignetController.php b/modules/custom/beignet/src/Controller/BeignetController.php
index 1b89a32..882e1cc 100644
--- a/modules/custom/beignet/src/Controller/BeignetController.php
+++ b/modules/custom/beignet/src/Controller/BeignetController.php
@@ -23,11 +23,6 @@ public static function create(ContainerInterface $container)
 
     public function eatBeignets($number)
     {
-        if ($number === null) {
-            $config = $this->configFactory->get('beignet.settings');
-            $number = $config->get('default_number');
-        }
-
         $bakery = new BeignetBakery();
         $beignets = $bakery->bakeBeignets($number);
 
diff --git a/modules/custom/beignet/src/Service/BeignetBakery.php b/modules/custom/beignet/src/Service/BeignetBakery.php
index 8580495..c03e71a 100644
--- a/modules/custom/beignet/src/Service/BeignetBakery.php
+++ b/modules/custom/beignet/src/Service/BeignetBakery.php
@@ -6,6 +6,11 @@ class BeignetBakery
 {
     public function bakeBeignets($number)
     {
+        if ($number === null) {
+            $config = $this->configFactory->get('beignet.settings');
+            $number = $config->get('default_number');
+        }
+
         $flavors = ['chocolate', 'plain', 'strawberry-filled'];
         $beignets = [];
         for ($i = 0; $i <= $number; $i++) {
