diff --git a/modules/custom/beignet/beignet.routing.yml b/modules/custom/beignet/beignet.routing.yml
index e1da5c6..6b775f4 100644
--- a/modules/custom/beignet/beignet.routing.yml
+++ b/modules/custom/beignet/beignet.routing.yml
@@ -2,7 +2,7 @@ beignet_eat:
     path: /beignets/eat/{number}
     defaults:
         _controller: Drupal\beignet\Controller\BeignetController::eatBeignets
-        number: 10
+        number: null
     requirements:
         _permission: 'access content'
 
diff --git a/modules/custom/beignet/src/Controller/BeignetController.php b/modules/custom/beignet/src/Controller/BeignetController.php
index 442c3cf..acfaae0 100644
--- a/modules/custom/beignet/src/Controller/BeignetController.php
+++ b/modules/custom/beignet/src/Controller/BeignetController.php
@@ -2,10 +2,30 @@
 
 namespace Drupal\beignet\Controller;
 
-class BeignetController
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class BeignetController implements ContainerInjectionInterface
 {
+    private $configFactory;
+
+    public function __construct($configFactory)
+    {
+        $this->configFactory = $configFactory;
+    }
+
+    public static function create(ContainerInterface $container)
+    {
+        return new self($container->get('config.factory'));
+    }
+
     public function eatBeignets($number)
     {
+        if ($number === null) {
+            $config = $this->configFactory->get('beignet.settings');
+            $number = $config->get('default_number');
+        }
+
         return [
             '#type' => 'markup',
             '#markup' => sprintf('NOM %s beignets', $number),
