diff --git a/modules/custom/beignet/src/Controller/BeignetController.php b/modules/custom/beignet/src/Controller/BeignetController.php
index 6763478..b5dfac5 100644
--- a/modules/custom/beignet/src/Controller/BeignetController.php
+++ b/modules/custom/beignet/src/Controller/BeignetController.php
@@ -2,6 +2,7 @@
 
 namespace Drupal\beignet\Controller;
 
+use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
@@ -9,7 +10,7 @@ class BeignetController implements ContainerInjectionInterface
 {
     private $configFactory;
 
-    public function __construct($configFactory)
+    public function __construct(ConfigFactoryInterface $configFactory)
     {
         $this->configFactory = $configFactory;
     }
