diff --git a/modules/custom/beignet/src/Controller/BeignetController.php b/modules/custom/beignet/src/Controller/BeignetController.php
index bdd25f2..a20eb24 100644
--- a/modules/custom/beignet/src/Controller/BeignetController.php
+++ b/modules/custom/beignet/src/Controller/BeignetController.php
@@ -23,7 +23,7 @@ public static function create(ContainerInterface $container)
 
     public function eatBeignets($number)
     {
-        $bakery = new BeignetBakery();
+        $bakery = new BeignetBakery($this->configFactory);
         $beignets = $bakery->bakeBeignets($number);
 
         return [
diff --git a/modules/custom/beignet/src/Service/BeignetBakery.php b/modules/custom/beignet/src/Service/BeignetBakery.php
index c03e71a..5ea9549 100644
--- a/modules/custom/beignet/src/Service/BeignetBakery.php
+++ b/modules/custom/beignet/src/Service/BeignetBakery.php
@@ -2,8 +2,17 @@
 
 namespace Drupal\beignet\Service;
 
+use Drupal\Core\Config\ConfigFactoryInterface;
+
 class BeignetBakery
 {
+    private $configFactory;
+
+    public function __construct(ConfigFactoryInterface $configFactory)
+    {
+        $this->configFactory = $configFactory;
+    }
+
     public function bakeBeignets($number)
     {
         if ($number === null) {
