diff --git a/modules/custom/beignet/src/Controller/BeignetController.php b/modules/custom/beignet/src/Controller/BeignetController.php
index 1aff27a..a1cf0df 100644
--- a/modules/custom/beignet/src/Controller/BeignetController.php
+++ b/modules/custom/beignet/src/Controller/BeignetController.php
@@ -9,7 +9,7 @@ class BeignetController extends ControllerBase
     public function eatBeignets($number)
     {
         if ($number === null) {
-            $config = $this->config('beignet.settings');
+            $config = $this->configFactory->get('beignet.settings');
             $number = $config->get('default_number');
         }
 
